docs/hub/how-to-inference.md

```diff
@@ -0,0 +1,44 @@
+# How to integrate the Inference API in your library
+
+The Inference API provides fast inference for your hosted models. Integration with the Hub allows you to access the Inference API programmatically. This guide will show you how to make calls to the Inference API from your library. For more detailed information, refer to the [Inference API documentation](https://api-inference.huggingface.co/docs/python/html/index.html).
+
+Begin by creating an instance of the `InferenceAPI` with a specific model repository ID:
+
+```python
+>>> from huggingface_hub.inference_api import InferenceApi
+>>> inference = InferenceApi(repo_id="bert-base-uncased", token=API_TOKEN)
+```
+
+The pipeline is determined from the metadata in the model card and configuration files (see [here](https://huggingface.co/bert-base-uncased/blob/main/config.json) for an example configuration file). For example, when using the [bert-base-uncased](https://huggingface.co/bert-base-uncased) model, the Inference API can automatically infer that this model should be used for a `fill-mask` task.
```

Comment:
Maybe it's also worth linking to https://huggingface.co/docs/hub/main#how-is-a-models-type-of-inference-api-and-widget-determined which provides useful info